src/backend/cdb/cdbmutate.c

```diff
@@ -2405,21 +2425,29 @@ shareinput_mutator_xslice_1(Node *node, PlannerInfo *root, bool fPop)
 	{
 		Motion	   *motion = (Motion *) plan;
 
-		shareinput_pushmot(ctxt, motion->motionID);
+		shareinput_pushmot(ctxt, motion);
 		return true;
 	}
 
 	if (IsA(plan, ShareInputScan))
 	{
 		ShareInputScan *sisc = (ShareInputScan *) plan;
 		int			motId = shareinput_peekmot(ctxt);
+		int			qds = shareinput_peekqds(ctxt);
 		Plan	   *shared = plan->lefttree;
 
-		Assert(sisc->scan.plan.flow);
-		if (sisc->scan.plan.flow->flotype == FLOW_SINGLETON)
+		if (qds == 1)
```

Comment:
done

Reply:
The value of such comments is almost zero. Do you think it's correctly described to somebody who will look at code in future?

Reply:
removed comments

Reply:
removed or replaced?

Reply:
comment about logic was before function https://github.com/arenadata/gpdb/pull/498/files#diff-a4950bc4a9e9552824f7ba4f6bf300b64335fb668074e51f7b9ee3ac30fe7a78L2388

Reply:
Does this line explain why we have such condition here?

Reply:
We're going in circles.
We had a voice talk about why it's bad that there no comment here and why we need it.
Instead of placing a comment here, we prefer to find 1000 and 1 reason to not do this. Why you afraid to do this?